Patches/Midworld Weaponry Expanded/Industrial_Ranged.xml

```diff
@@ -0,0 +1,260 @@
+						<power>5</power>
+						<cooldownTime>2.02</cooldownTime>
+						<armorPenetrationBlunt>1.630</armorPenetrationBlunt>
+						<linkedBodyPartsGroup>Barrel</linkedBodyPartsGroup>
+					</li>
+					<li Class="CombatExtended.ToolCE">
+						<label>muzzle</label>
+						<capacities>
+							<li>Poke</li>
+						</capacities>
+						<power>8</power>
+						<cooldownTime>1.55</cooldownTime>
+						<armorPenetrationBlunt>2.755</armorPenetrationBlunt>
+						<linkedBodyPartsGroup>Muzzle</linkedBodyPartsGroup>
+					</li>
+				</tools>
+			</value>
+		</li>
+
+
+		<li Class="PatchOperationReplace">
+			<xpath>/Defs/ThingDef[defName="Gun_AdvancedGrenadeLauncher"]/tools</xpath>
+			<value>
+				<tools>
+					<li Class="CombatExtended.ToolCE">
+						<label>barrel</label>
+						<capacities>
+							<li>Blunt</li>
+						</capacities>
+						<power>10</power>
+						<cooldownTime>2.44</cooldownTime>
+						<armorPenetrationBlunt>3.5</armorPenetrationBlunt>
+						<linkedBodyPartsGroup>Barrel</linkedBodyPartsGroup>
+					</li>
+				</tools>
+			</value>
+		</li>
+
+		
+		
+		
+		<!-- === Advanced Assault Rifle === -->
+		<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+			<defName>Gun_AdvancedRifle</defName>
+			<statBases>
+				<WorkToMake>40000</WorkToMake>
+				<Mass>4.00</Mass>
+				<Bulk>7.50</Bulk>
+				<SwayFactor>1.24</SwayFactor>
+				<ShotSpread>0.11</ShotSpread>
+				<SightsEfficiency>1.2</SightsEfficiency>
+				<RangedWeapon_Cooldown>0.34</RangedWeapon_Cooldown>
+			</statBases>
+			<costList>
+				<Steel>55</Steel>
+				<Chemfuel>15</Chemfuel>
+				<Gold>7</Gold>
+				<ComponentIndustrial>6</ComponentIndustrial>
+			</costList>
+			<Properties>
+				<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+				<hasStandardCommand>True</hasStandardCommand>
+				<defaultProjectile>Bullet_545x39mmSoviet_FMJ</defaultProjectile>
+				<burstShotCount>6</burstShotCount>
+				<ticksBetweenBurstShots>6</ticksBetweenBurstShots>
+				<warmupTime>1.1</warmupTime>
+				<range>48</range>
+				<soundCast>Shot_AssaultRifle</soundCast>
+				<soundCastTail>GunTail_Medium</soundCastTail>
+				<muzzleFlashScale>9</muzzleFlashScale>
+				<recoilAmount>2.12</recoilAmount>
+			</Properties>
+			<AmmoUser>
+				<magazineSize>60</magazineSize>
+				<reloadTime>4</reloadTime>
+				<ammoSet>AmmoSet_545x39mmSoviet</ammoSet>
+			</AmmoUser>
+			<FireModes>
+				<aiAimMode>AimedShot</aiAimMode>
+				<aiUseBurstMode>True</aiUseBurstMode>
+				<aimedBurstShotCount>3</aimedBurstShotCount>
+			</FireModes>
+			<weaponTags>
+				<li>IndustrialGunAdvanced</li>
+			</weaponTags>
+			<!-- Required Research Rework -->
+			<researchPrerequisite>PrecisionRifling</researchPrerequisite>
+		</li>
+
+	
+		<!-- === Compact Rifle === -->
+		<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+			<defName>Gun_CompactRifle</defName>
+			<statBases>
+				<WorkToMake>20000</WorkToMake>
+				<Mass>2.65</Mass>
+				<Bulk>4.00</Bulk>
+				<SwayFactor>1.15</SwayFactor>

```

Comment:
This is probably a bit too low for a heavier weapon that doesn't have a stock. I know it's a two-handed weapon, but for stat purposes, I'd calculate it as a one-handed weapon and perhaps reduce it from there a bit.